server/api/routes.md

```diff
@@ -0,0 +1,49 @@
+**USER: (/api/user)**
+
+* register guest user (POST) - occurs as soon as any user visits site (any page). Sets cookie (sessionId)
+* create user account (POST)
+* authenticate registered user (POST) - sets cookie (some version of userId)
+* retrieve single user, or all users (GET)
+* update user (PUT)
+* delete user account (DELETE)
+
+**CART/CART-ITEM: (/api/cart)**
+* create new cart (POST) - use userId (new cart should be created as soon as new user enters app)
+* retrieve single cart, or all carts (GET)
+* add to cart (POST) - creates new CartItem, uses productId of selected product, cartId
+* retrieve single CartItem, or all CartItem(s) (GET)
+* edit cart item (PUT)
+* delete cart item (DELETE)
+* clear cart (PUT)
+
+**PRODUCT: (/api/products)**
+* add new product (POST) - this should create new price in the process
+* edit product (PUT) - this will be used whenever a user adds a product to cart or updates quantity, as well as admin purposes
+* retrieve individual product, or all products (GET) - add queries to sort and filter (e.g. product in category)
+* delete product (DELETE) - admin only!
+
+**PRICE: (/api/price)**
+* create new price (POST)
+* retrieve single price, or all prices (GET)
+* update price (PUT)
```

Comment:
I'm thinking we should change the relationship between PRODUCT and PRICE models. I think we should make sure that PRICE belongs to  PRODUCT so if  I  update the price, it only updates for one given  product.

Reply:
Looks like it's currently set up so that Price has a productId. I think it's fine this  way...?

Reply:
Do we really need Price model separately? if yes please let me know what is the thought process.

Reply:
It was recommended in some database design examples as a way to simplify updating prices. 
We can collapse it into the  PRODUCT model  to start if it's easier.

Reply:
yes.